catalogs/catalog-lakehouse-iceberg/src/main/java/com/datastrato/gravitino/catalog/lakehouse/iceberg/web/metrics/IcebergMetricsManager.java

```diff
@@ -0,0 +1,169 @@
+
+  enum MetricsStorageType {
+    EMPTY
+  };
+
+  private final IcebergMetricsFormatter icebergMetricsFormatter;
+  private final IcebergMetricsStorage storage;
+  private final boolean asyncMode = true;
+  private final BlockingQueue<MetricsReport> queue;
+  private final Thread metricsWriterThread;
+  private final int retainDays;
+
+  private volatile boolean isClosed = false;
+  private ScheduledExecutorService scheduledThreadPoolExecutor;
+
+  public IcebergMetricsManager(IcebergConfig icebergConfig) {
+    icebergMetricsFormatter = new IcebergMetricsFormatter();
+    String metricsStorage = icebergConfig.get(IcebergConfig.ICEBERG_METRICS_STORAGE);
+    if (StringUtils.isBlank(metricsStorage)) {
+      metricsStorage = "EMPTY";
+    }
+    MetricsStorageType type = MetricsStorageType.valueOf(metricsStorage.toUpperCase());
+    LOG.info("Iceberg metrics storage: {}", type);
+    switch (type) {
+      case EMPTY:
+        storage = new EmptyMetricsStorage();
+        break;
+      default:
+        throw new RuntimeException("Unknown Iceberg metrics storage: " + type);
+    }
+
+    retainDays = icebergConfig.get(IcebergConfig.ICEBERG_METRICS_STORAGE_RETAIN_DAYS);
+    if (retainDays > 0) {
+      scheduledThreadPoolExecutor =
+          new ScheduledThreadPoolExecutor(
+              1,
+              new ThreadFactoryBuilder()
+                  .setDaemon(true)
+                  .setNameFormat("Iceberg-metrics-cleaner")
+                  .setUncaughtExceptionHandler(
+                      (t, e) -> LOG.error("Uncaught exception in thread {}", t, e))
+                  .build());
+    }
+
+    if (asyncMode) {
+      queue = new LinkedBlockingQueue(1000);
+      metricsWriterThread =
+          new Thread(
+              () -> {
+                while (Thread.currentThread().isInterrupted() == false) {
+                  MetricsReport metricsReport;
+                  try {
+                    metricsReport = queue.poll(1, TimeUnit.MINUTES);
+                  } catch (InterruptedException e) {
+                    LOG.warn("Iceberg Metrics writer thread is interrupted");
+                    return;
+                  }
+                  if (metricsReport != null) {
+                    doSave(metricsReport);
+                  }
+                }
+
+                MetricsReport metricsReport = queue.poll();
+                while (metricsReport != null) {
+                  logMetrics(
+                      "Drop Iceberg metrics because it's time to close metrics storage.",
+                      metricsReport);
+                  metricsReport = queue.poll();
+                }
+              });
+      metricsWriterThread.setName("Iceberg-metrics-writer");
+      metricsWriterThread.setDaemon(true);
+    }
+  }
+
+  private void logMetrics(String message, MetricsReport metricsReport) {
+    LOG.info("{} {}", message, icebergMetricsFormatter.toJsonOrToString(metricsReport));
```

Comment:
Looks like this `IcebergMetricsFormatter`  is only used here, Is it necessary to have a class to define this?

Reply:
IcebergMetricsFormatter is designed to provide variable format for other storages like `KVStorage` or `SQLStorage`